provider/core/src/response.rs

```diff
@@ -951,10 +920,10 @@ impl<M> TryFrom<DataResponse<M>> for DataPayload<M>
 where
     M: DynamicDataMarker,
 {
-    type Error = DataError;
+    type Error = core::convert::Infallible;
 
     fn try_from(response: DataResponse<M>) -> Result<Self, Self::Error> {
-        response.take_payload()
+        Ok(response.payload)
```

Comment:
Thought: Just make this `impl From`

Reply:
I'll remove it, there's no point of a `From` that just accesses a field.